test/02-investigate.t

```diff
@@ -61,6 +86,51 @@ is "$rc" 0 'success regardless of actually triggered jobs'
 is "$out" "Job already has a clone, skipping investigation. Use the env variable 'force=true' to trigger investigation jobs"
 
 rc=0
-out=$(force=true investigate 27 2>&1) || rc=$?
-is "$rc" 0 'still success'
+out=$(force=true investigate 28 2>&1) || rc=$?
+is "$rc" 0 'still success when job is skipped (because of exclude_no_group)'
 like "$out" "exclude_no_group is set, skipping investigation"
+
+rc=0
+out=$(investigate 30 2>&1) || rc=$?
+is "$rc" 142 'investigation postponed because other job in cluster is not done'
+like "$out" "Postponing to investigate job 30: waiting until pending dependencies have finished"
+
+rc=0
+out=$(force=true investigate 31 2>&1) || rc=$?
+is "$rc" 0 'success when job is skipped (because of exclude_no_group annd job w/o group)'
+like "$out" 'Job w/o job group, \$exclude_no_group is set, skipping investigation'
+
+# test syncing via investigation comment; we're first
+rc=0
+out=$(force=true sync_via_investigation_comment 31 30 2>&1) || rc=$?
+is "$rc" 255 'do not skip if we own first investigation comment'
+like "$out" '1234' 'comment ID returned'
+
+# test syncing via investigation comment; we're second
+rc=0
+out=$(force=true sync_via_investigation_comment 32 32 2>&1) || rc=$?
+is "$rc" 0 'skip with success if we do not own first investigation comment'
+like "$out" '' 'no output when skipping'
+
+# delete certain files used to trace whether API calls happened
+for trace_file in comment_1234_updated comment_for_job_31_created comment_1234_deleted; do
+    [[ -f $trace_file ]] && unlink "$trace_file"
+done
+
+# test finalizing investigation comment when no investigation jobs were needed
+rc=0
+out=$(force=true finalize_investigation_comment 31 30 1234 '' 2>&1) || rc=$?
+is "$rc" 0 'success if no investigation jobs needed to be created after all'
+[[ -f comment_1234_deleted ]] && comment_1234_deleted=1 || comment_1234_deleted=0
```

Comment:
this could be

```suggestion
[[ ! -f comment_1234_deleted ]]; comment_1234_deleted=$?
```

Reply:
It somehow doesn't work; if the condition is false this is considered an error and the test stops. Only using `set +e` before would make it work.

Putting
```
some-function() {
    return 255
}
another-function() {
    foo=$(some-function ); rc=$?
    [[ $rc != 255 ]] && return $rc
    return 0
}
another-function
```
there gives me the same problem.

But I don't really understand it because why is then
```
    dependency_data=$(query_dependency_data_or_postpone "$id" "$job_data"); rc=$?
    [[ $rc != 255 ]] && return $rc
```
working? It also works if I explicitly invoke `set -e` before. And I'm pretty sure that it works here because if it wouldn't work the test would fail and the script would always skip the investigation.

Reply:
well, nevermind then.